packages/fiori-freestyle-writer/src/packageConfig.ts

```diff
@@ -28,21 +28,27 @@ export function getPackageJsonTasks({
     startFile?: string;
     localStartFile?: string;
 }): { start: string; 'start-local': string; 'start-noflp': string; 'start-mock'?: string } {
-    const sapClientParam = sapClient ? `?sap-client=${sapClient}` : '';
-    let urlParam = [sapClientParam, 'sap-ui-xx-viewCache=false'].filter((param) => !!param).join('&');
-    urlParam = urlParam ? `?${urlParam}` : '';
-    const params = `${urlParam}${flpAppId ? `#${flpAppId}` : ''}`;
+
+    const addParams = [['sap-ui-xx-viewCache', 'false']];
+    if (sapClient) {
+        addParams.push([`sap-client`, `${sapClient}`]);
+    }
+    const searchParams = new URLSearchParams(addParams);
+    const flpParams = `?${searchParams}${flpAppId ? `#${flpAppId}` : ''}`;
```

Comment:
Thanks Iain, I forgot to remove '?' in ?sap-client=${sapClient}

If this disable cache param is removed in the future for some reason, searchParams could be empty.  It is better to check searchParams is empty and decide if we add ? instead of always having it in the string template.

Reply:
@slin-sap Should I make the cache param configurable via an additional function param, is it likely to be toggled?

Reply:
@IainSAP We never know if we might turn off this cache param and left with a single optional param again. Despite this, I think it is better to separate the duties. 

E.g.

```
// Complete build of query params of url
searchParams = new URLSearchParams(addParams);
searchParams = searchParams ? searchParams : '';
// Complete build of hash fragment of url
hashFragment = flpAppId ? `#${flpAppId}` : '';


// Building the full params of the url can be simply concatenate 
// the query part and the hash part together, since previously code section already handled the empty string cases.
const flpParams = `${searchParams}${hashFragment}`;

// Finally, in the code section for building various start commands, 
// no need to think about if flpParams or searchParams have provided ? / # prefix. 
// Only need to focus on concatenating the url param parts to the html page path. 
const startLocalCommand = `fiori run --config ./ui5-local.yaml --open '${
    localStartFile || 'test/flpSandbox.html'
}${flpParams}'`;
    
const startNoFlpCommand = localOnly
  ? `echo \\"${t('info.mockOnlyWarning')}\\"`
  : `fiori run --open '${'index.html'}{searchParams}'`;
```

Reply:
Sorry , I dont see an issue with the current implementation. Can you take over the PR please?